src/OmniSharp.DotNetTest/Services/DebugTestsInContextService.cs

```diff
@@ -0,0 +1,45 @@
+#nullable enable
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+using OmniSharp.DotNetTest.Models;
+using OmniSharp.Eventing;
+using OmniSharp.Mef;
+using OmniSharp.Services;
+
+namespace OmniSharp.DotNetTest.Services
+{
+    [Shared]
+    [OmniSharpHandler(OmniSharpEndpoints.V2.DebugTestsInContextGetStartInfo, LanguageNames.CSharp)]
+    internal class DebugTestsInContextService : BaseTestService,
+        IRequestHandler<DebugTestsInContextGetStartInfoRequest, DebugTestGetStartInfoResponse>
+    {
+        private readonly DebugSessionManager _debugSessionManager;
+
+        [ImportingConstructor]
+        public DebugTestsInContextService(DebugSessionManager debugSessionManager, OmniSharpWorkspace workspace, IDotNetCliService dotNetCli, IEventEmitter eventEmitter, ILoggerFactory loggerFactory)
+            : base(workspace, dotNetCli, eventEmitter, loggerFactory)
+        {
+            _debugSessionManager = debugSessionManager;
+        }
+
+        public Task<DebugTestGetStartInfoResponse> Handle(DebugTestsInContextGetStartInfoRequest request)
+        {
+            var testManager = CreateTestManager(request.FileName);
+            if (testManager.IsConnected)
+            {
+                _debugSessionManager.StartSession(testManager);
+                return _debugSessionManager.DebugGetStartInfoAsync(
+                    request.Line, request.Column, Workspace.GetDocument(request.FileName),
+                    request.RunSettings, request.TargetFrameworkVersion,
+                    CancellationToken.None);
+            }
+
+            throw new InvalidOperationException("The debugger could not be started");
```

Comment:
These seem unfortunate, especially for the `InContext` version of the command, but the existing `DebugTestGetStartInfoResponse` doesn't have a place for a failure and failure message. I would have modified that to add a field, but I'm unsure if that will break existing clients. Is that a safe change to make, or should I add a new response that adds a failure/reason to it?

Reply:
I'm not really sure whether this would be a problem for other clients. Maybe @filipw knows definitively.

Reply:
it should be fine - additive changes are generally fine since it's json based.
moreover, the debug tests protocl is - I am pretty sure - only used by VS Code extension since the debugger is proprietary